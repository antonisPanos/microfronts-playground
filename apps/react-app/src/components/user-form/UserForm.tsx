import React, {Fragment, FunctionComponent, ReactElement} from 'react';
import type {MaestroWindow, UserDetails} from '../../../../../libs/models/types';
import type {BehaviorSubject} from 'rxjs';
import type {RouteComponentProps} from '@reach/router';
import {hot} from 'react-hot-loader';

export const UserForm: FunctionComponent<RouteComponentProps<unknown>> = (): ReactElement => {
  const user$ = React.useRef<BehaviorSubject<UserDetails> | undefined>(
    (window as MaestroWindow).maestro?.user$
  );
  const [user, setUser] = React.useState<UserDetails>({name: ''});

  React.useEffect(() => {
    const subscription = user$.current?.subscribe(value => {
      setUser(value);
    });

    return () => {
      subscription?.unsubscribe();
    };
  }, [user$.current?.getValue()]);

  return (
    <Fragment>
      <h3>User</h3>
      <input
        type={'text'}
        value={user?.name}
        onChange={({target}) => user$?.current?.next({name: target.value})}
      />
    </Fragment>
  );
};

declare let module: Record<string, unknown>;

export default hot(module)(UserForm);
