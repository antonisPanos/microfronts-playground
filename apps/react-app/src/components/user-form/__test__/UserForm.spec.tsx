import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {UserForm} from '../UserForm';

describe('RouteInfo', () => {
  it('should render correctly', () => {
    const mainComp: ShallowWrapper = shallow(<UserForm />);

    expect(mainComp.debug()).toMatchSnapshot();
  });
});
