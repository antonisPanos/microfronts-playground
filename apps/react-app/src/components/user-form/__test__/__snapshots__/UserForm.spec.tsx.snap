// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteInfo should render correctly 1`] = `
"<Fragment>
  <h3>
    User
  </h3>
  <input type=\\"text\\" value=\\"\\" onChange={[Function: onChange]} />
</Fragment>"
`;
